### PR TITLE
Improve guided mode editor

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?

- [x] New "Full"-guided-mode that hides all settings and toolbars for dynamic services. It is meant to be used for linear-only pipelines.
- [x] For interactive services, show interactive state in breadcrumb instead of output's state.
- [x] Edit guided mode from breadcrumbs:
  - Only available when the pipeline is linear WIP
  - New nodes can be added at any position. Connections will be made automatically.
  - Existing nodes can be deleted. Connections will be bypassed.
  - Nodes can be shown/hidden
- [x] Everything related to the Guided Mode moved to the Navigation Bar

breadcrumbs

#### Bonus:
- [x] Favicon depends on product
- [x] New osparc logo
- [x] Fixed always visible delete button
- [x] e2e: When selecting a service for starting a new study, we now check the label instead of picking the first in the list.

## Related issue/s
fixes https://github.com/ITISFoundation/osparc-issues/issues/495

### CODE REVIEW

**Review by GitHK — COMMENTED** [review]
Screenshot 2021-08-06 at 09 42 35

The above  is created by adding a file picker to the last service (rightmost blu add button) from the guided mode.

---

Screenshot 2021-08-06 at 09 41 36

In a similar way as before just add a file picker between 2 nodes.

**Comment by odeimaiz** [comment]
> Screenshot 2021-08-06 at 09 42 35
> 
> The above is created by adding a file picker to the last service (rightmost blu add button) from the guided mode.
> 
> Screenshot 2021-08-06 at 09 41 36
> 
> In a similar way as before just add a file picker between 2 nodes.

Fixed, plus button is hidden if nothing that can be connected.

**Review by ignapas — DISMISSED** [review]
I found a weird behavior where the edit/accept buttons disappear when I add a node between two nodes where a middle node was removed from the guided mode
Peek 2021-08-09 18-27

File: services/web/client/source/class/osparc/component/widget/NodesSlidesTree.js [hunk 1]
```diff
@@ -185,41 +168,75 @@ qx.Class.define("osparc.component.widget.NodesSlidesTree", {
...
+      children.sort((a, b) => {
+        const aPos = a.getPosition();
+        const bPos = b.getPosition();
+        if (aPos === -1) {
+          return 1;
+        }
+        if (bPos === -1) {
+          return -1;
+        }
+        return aPos - bPos;
```
[the remark this answers is not in the document]
Comment (odeimaiz) [reply services/web/client/source/class/osparc/component/widget/NodesSlidesTree.js:216]: Yep, it needs to return less than zero, greater than zero or zero.

File: services/web/client/source/class/osparc/component/service/ServiceButtonSmall.js [hunk 2]
```diff
@@ -68,8 +68,13 @@ qx.Class.define("osparc.component.service.ServiceButtonSmall", {
...
         this.addListener("mouseover", showHint);
-        this.addListener("mouseout", hideHint);
-        this.addListener("dbltap", hideHint);
+        [
+          "mouseout",
+          "dbltap",
+          "keypress"
+        ].forEach(e => {
+          this.addListener(e, hideHint);
+        });
```
[the remark this answers is not in the document]
Comment (odeimaiz) [reply services/web/client/source/class/osparc/component/service/ServiceButtonSmall.js:77]: I could make it four lines where one is a square bracket... still more clear to me that the listener to all those signals behaves the same.

**Comment by odeimaiz** [comment]
> I found a weird behavior where the edit/accept buttons disappear when I add a node between two nodes where a middle node was removed from the guided mode
> Peek 2021-08-09 18-27

The reason why it disappears is that the pipeline is not linear anymore. The question here is if we should also hide it when all the nodes in the pipeline are not part of the guided mode. Because in your case, when you add the fourth sleeper, should it go right after ```sleeper``` or right before ```sleeper_3``` ...

File: services/web/client/source/class/osparc/component/widget/NodesSlidesTree.js [hunk 3]
```diff
@@ -185,41 +168,75 @@ qx.Class.define("osparc.component.widget.NodesSlidesTree", {
...
-          const nodeId = child.getNodeId();
-          if (nodeId in initData) {
-            child.setPosition(initData[nodeId].position);
-            child.setSkipNode(false);
-          } else {
-            child.setPosition(-1);
-            child.setSkipNode(true);
-          }
-        });
+    __convertToModel: function(nodes) {
```
[the remark this answers is not in the document]
Comment (odeimaiz) [reply services/web/client/source/class/osparc/component/widget/NodesSlidesTree.js:171]: Done

**Review by sanderegg — APPROVED** [review]
unleash it!

**Review by mguidon — APPROVED** [review]

**Pull request merged**